### PR TITLE
[3.4] bpo-36216: Add check for characters in netloc that normalize to separators (GH-12201)

### PR DESCRIPTION
**Pull request by zooba**
https://bugs.python.org/issue36216

### CODE REVIEW

**Comment by zooba** [comment]
Ping @larryhastings (but wait for CI, since I didn't re-test locally with 3.4)

File: Lib/urllib/parse.py [hunk 1]
```diff
@@ -316,6 +316,21 @@ def _splitnetloc(url, start=0):
...
+def _checknetloc(netloc):
+    if not netloc or not any(ord(c) > 127 for c in netloc):
+        return
+    # looking for characters like \u2100 that expand to 'a/c'
+    # IDNA uses NFKC equivalence, so normalize for this check
+    import unicodedata
+    netloc2 = unicodedata.normalize('NFKC', netloc)
+    if netloc == netloc2:
+        return
+    _, _, netloc = netloc.rpartition('@') # anything to the left of '@' is okay
```
Comment (larryhastings) [inline Lib/urllib/parse.py:328]: I'm not asking you to change this, just a suggestion in the future: you might as well have spelled this

`netloc = netloc.rpartition('@')[2]`

**Comment by larryhastings** [comment]
Thanks for the backport!  Our cookies are once again safe.

**Comment by mcepl** [comment]
> Thanks for the backport! Our cookies are once again safe.

That sounds shockingly dirty …

**Pull request merged**